### DISCUSSION TITLE
Arduino-Pico included library priority

### DISCUSSION THREAD

**Discussion by FerGT50** [earlephilhower/arduino-pico#1945] [Q&A]
Hello,
while studying TinyUSB stuff, I encountered a problem: with Arduino IDE 1.8.19 under Windows 10, the compiler picks up a version of Adafruit_TinyUSB installed in my sketches folder ("library" subfolder) instead of the one included in Arduino-Pico (that is, C:\Users\<USERNAME>\AppData\Local\Arduino15\packages\rp2040\hardware\rp2040\3.6.3\libraries\Adafruit_TinyUSB_Arduino).
This causes various build errors.

I solved editing 

`C:\Users\<USERNAME>\AppData\Local\Arduino15\packages\rp2040\hardware\rp2040\3.6.3\libraries\Adafruit_TinyUSB_Arduino\library.properties
`

replacing 

`architecture=*`

with

`architecture=rp2040`

This way the Arduino-Pico version has higher priority (since it exactly matches the selected board architecture)

Maybe it could be the default for libraries included in the Arduino-Pico core?

**Comment by earlephilhower** [comment]
We use a completely unmodified library to keep things as simple and sane for me as possible.  So, you should be able to run any copy of the library, as long as the versions are close so that no API changes were made.  That also makes even changes like you're suggesting not possible.  For libraries native here, like SPI, I do believe I already specify only the rp2040 architecture.

What exactly was the problem you found?  Or were you just concerned that it was picking up the wrong copy?

**Comment by FerGT50** [comment]
The problem is it always picks up the wrong copy (I had to install Adafruit-TinyUSB, for other projects using other platforms), which gives build errors.
